controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1433,6 +1446,21 @@ func (r *BareMetalHostReconciler) doServiceIfNeeded(prov provisioner.Provisioner
 		dirty = true
 	}
 
+	if hfcDirty && started {
+		hfcStillDirty, err := r.saveHostFirmwareComponents(prov, info, hfc)
```

Comment:
nit: you can reuse hfcDirty since you no longer need its original value

Reply:
Thanks!